src/lib.rs

```diff
@@ -847,6 +857,215 @@ impl Config {
     }
 }
 
+trait Target {
+    fn add_cmake_defines(&self, _cmd: &mut Command, _config: &Config) {}
+
+    fn should_exclude_env_var(&self, _key: &OsStr, _value: &OsStr) -> bool {
+        false
+    }
+
+    fn filter_compiler_args(&self, _flags: &mut OsString) {}
+}
+
+fn get_target(target_triple: &str) -> Box<dyn Target> {
+    let target: Option<Box<dyn Target>>;
+    target = AppleTarget::new(target_triple)
+        .map(|apple_target| Box::new(apple_target) as Box<dyn Target>);
+    target.unwrap_or_else(|| Box::new(GenericTarget::new(target_triple)))
+}
+
+struct GenericTarget {}
+
+impl GenericTarget {
+    fn new(_target_triple: &str) -> GenericTarget {
+        GenericTarget {}
+    }
+}
+
+impl Target for GenericTarget {}
+
+struct AppleTarget {
+    rust_target: String,
+    rust_target_arch: String,
+    rust_target_platform: String,
+}
+
+impl AppleTarget {
+    fn new(target_triple: &str) -> Option<AppleTarget> {
+        let parts: Vec<&str> = target_triple.split('-').collect();
+        if parts.len() < 3 {
+            return None;
+        }
+
+        let rust_target_arch = parts[0];
+        let rust_target_vendor = parts[1];
+        let rust_target_platform = parts[2];
+
+        if rust_target_vendor != "apple" {
+            return None;
+        }
+        if rust_target_platform != "ios" && !rust_target_platform.starts_with("darwin") {
+            eprintln!(
+                "Warning: unknown Apple platform ({}) for target: {}",
+                rust_target_platform, target_triple
+            );
+            return None;
+        }
+
+        Some(AppleTarget {
+            rust_target: target_triple.to_owned(),
+            rust_target_arch: rust_target_arch.to_owned(),
+            rust_target_platform: rust_target_platform.to_owned(),
+        })
+    }
+
+    fn is_ios_target(&self) -> bool {
+        self.rust_target_platform == "ios"
+    }
+
+    fn is_osx_target(&self) -> bool {
+        self.rust_target_platform.starts_with("darwin")
+    }
+
+    fn cmake_target_arch(&self) -> Option<String> {
+        match self.rust_target_arch.as_str() {
+            "aarch64" => Some("arm64".to_owned()),
+            "armv7" => Some("armv7".to_owned()),
+            "armv7s" => Some("armv7s".to_owned()),
+            "i386" => Some("i386".to_owned()),
+            "x86_64" => Some("x86_64".to_owned()),
+            _ => {
+                eprintln!(
+                    "Warning: unknown architecture for target: {}",
```

Comment:
Should these warnings use the [`cargo:warning=MESSAGE`](https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargowarningmessage) format?